drjt/Animal/lib/Animal.pm

```diff
@@ -38,6 +38,77 @@ if you don't export anything, such as for a purely object-oriented module.
 
 =head1 SUBROUTINES/METHODS
 
+=head2 new
+
+Constructor, takes a name as a parameter
+
+=cut
+
+sub new {
+  my $class = shift;
+  my $name = shift;
+  my $self = { name => $name, colour => $class->default_colour() };
+  bless $self, $class;
+  return $self;
+}
+
+=head2 default_colour
+
+Default attribute for the object
+
+=cut
+
+sub default_colour { 'beige' }
+
+=head2 set_name
+
+A setter
+
+=cut
+
+sub set_name {
+  my $self = shift;
+  my $name = shift;
+  if ( ! ref $self ) {
+    die "Class has not been instantiated, Call a constructor first!\n";
+  }
```

Comment:
Huh, must have missed that bit. But yeah, that's what the new-line is for, it prevents the `die` from including the module and line number - so not as useful, but it doesn't expose any workings.